File: TyreVibes/Core/Component/PremiumBadge.swift
import SwiftUI

/// Badge che indica lo stato Premium dell'utente
struct PremiumBadge: View {
    let size: BadgeSize
    let showLabel: Bool

    init(size: BadgeSize = .medium, showLabel: Bool = true) {
        self.size = size
        self.showLabel = showLabel
    }

    var body: some View {
        HStack(spacing: size.spacing) {
            Image(systemName: "crown.fill")
                .font(.system(size: size.iconSize))
                .foregroundColor(.yellow)

            if showLabel {
                Text("Premium")
                    .font(.customFont(size: size.fontSize, weight: .semibold))
                    .foregroundColor(.white)
            }
        }
        .padding(.horizontal, size.horizontalPadding)
        .padding(.vertical, size.verticalPadding)
        .background(
            LinearGradient(
                colors: [Color.yellow.opacity(0.3), Color.orange.opacity(0.3)],
                startPoint: .leading,
                endPoint: .trailing
            )
        )
        .cornerRadius(size.cornerRadius)
        .overlay(
            RoundedRectangle(cornerRadius: size.cornerRadius)
                .stroke(
                    LinearGradient(
                        colors: [Color.yellow, Color.orange],
                        startPoint: .leading,
                        endPoint: .trailing
                    ),
                    lineWidth: 1
                )
        )
    }

    enum BadgeSize {
        case small
        case medium
        case large

        var iconSize: CGFloat {
            switch self {
            case .small: return 12
            case .medium: return 16
            case .large: return 20
            }
        }

        var fontSize: CGFloat {
            switch self {
            case .small: return 10
            case .medium: return 14
            case .large: return 16
            }
        }

        var horizontalPadding: CGFloat {
            switch self {
            case .small: return 8
            case .medium: return 12
            case .large: return 16
            }
        }

        var verticalPadding: CGFloat {
            switch self {
            case .small: return 4
            case .medium: return 6
            case .large: return 8
            }
        }

        var cornerRadius: CGFloat {
            switch self {
            case .small: return 8
            case .medium: return 10
            case .large: return 12
            }
        }

        var spacing: CGFloat {
            switch self {
            case .small: return 4
            case .medium: return 6
            case .large: return 8
            }
        }
    }
}

/// Badge per indicare funzionalità bloccate per utenti free
struct LockedFeatureBadge: View {
    let size: PremiumBadge.BadgeSize
    let showLabel: Bool

    init(size: PremiumBadge.BadgeSize = .medium, showLabel: Bool = true) {
        self.size = size
        self.showLabel = showLabel
    }

    var body: some View {
        HStack(spacing: size.spacing) {
            Image(systemName: "lock.fill")
                .font(.system(size: size.iconSize))
                .foregroundColor(.yellow)

            if showLabel {
                Text("Premium")
                    .font(.customFont(size: size.fontSize, weight: .semibold))
                    .foregroundColor(.white.opacity(0.8))
            }
        }
        .padding(.horizontal, size.horizontalPadding)
        .padding(.vertical, size.verticalPadding)
        .background(Color.white.opacity(0.1))
        .cornerRadius(size.cornerRadius)
        .overlay(
            RoundedRectangle(cornerRadius: size.cornerRadius)
                .stroke(Color.yellow.opacity(0.5), lineWidth: 1)
        )
    }
}

/// Badge per mostrare il limite raggiunto
struct LimitReachedBadge: View {
    let current: Int
    let max: Int

    var body: some View {
        HStack(spacing: 6) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 12))
                .foregroundColor(.orange)

            Text("\(current)/\(max)")
                .font(.customFont(size: 12, weight: .semibold))
                .foregroundColor(.white)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(Color.orange.opacity(0.2))
        .cornerRadius(8)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color.orange.opacity(0.5), lineWidth: 1)
        )
    }
}

#Preview("Premium Badge") {
    ZStack {
        Color.customBackgroundColor
            .ignoresSafeArea()

        VStack(spacing: 20) {
            PremiumBadge(size: .small)
            PremiumBadge(size: .medium)
            PremiumBadge(size: .large)

            PremiumBadge(size: .small, showLabel: false)
            PremiumBadge(size: .medium, showLabel: false)

            LockedFeatureBadge(size: .medium)
            LockedFeatureBadge(size: .small, showLabel: false)

            LimitReachedBadge(current: 2, max: 2)
        }
    }
}

File: android/app/src/main/java/it/tyrevibes/app/ui/components/PremiumBadge.kt
package it.tyrevibes.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import it.tyrevibes.app.ui.theme.SoraFontFamily

enum class BadgeSize {
    SMALL, MEDIUM, LARGE;

    val iconSize: Int
        get() = when (this) {
            SMALL -> 12
            MEDIUM -> 16
            LARGE -> 20
        }

    val fontSize: Int
        get() = when (this) {
            SMALL -> 10
            MEDIUM -> 14
            LARGE -> 16
        }

    val horizontalPadding: Int
        get() = when (this) {
            SMALL -> 8
            MEDIUM -> 12
            LARGE -> 16
        }

    val verticalPadding: Int
        get() = when (this) {
            SMALL -> 4
            MEDIUM -> 6
            LARGE -> 8
        }

    val cornerRadius: Int
        get() = when (this) {
            SMALL -> 8
            MEDIUM -> 10
            LARGE -> 12
        }

    val spacing: Int
        get() = when (this) {
            SMALL -> 4
            MEDIUM -> 6
            LARGE -> 8
        }
}

/**
 * Premium Badge - Indica lo stato Premium dell'utente
 */
@Composable
fun PremiumBadge(
    modifier: Modifier = Modifier,
    size: BadgeSize = BadgeSize.MEDIUM,
    showLabel: Boolean = true
) {
    Row(
        modifier = modifier
            .background(
                brush = Brush.linearGradient(
                    colors = listOf(
                        Color(0xFFFFD700).copy(alpha = 0.3f),
                        Color(0xFFFF8C00).copy(alpha = 0.3f)
                    ),
                    start = Offset.Zero,
                    end = Offset.Infinite
                ),
                shape = RoundedCornerShape(size.cornerRadius.dp)
            )
            .border(
                width = 1.dp,
                brush = Brush.linearGradient(
                    colors = listOf(
                        Color(0xFFFFD700),
                        Color(0xFFFF8C00)
                    )
                ),
                shape = RoundedCornerShape(size.cornerRadius.dp)
            )
            .padding(
                horizontal = size.horizontalPadding.dp,
                vertical = size.verticalPadding.dp
            ),
        horizontalArrangement = Arrangement.spacedBy(size.spacing.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = Icons.Default.Star,
            contentDescription = "Premium",
            tint = Color(0xFFFFD700),
            modifier = Modifier.size(size.iconSize.dp)
        )

        if (showLabel) {
            Text(
                text = "Premium",
                fontFamily = SoraFontFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = size.fontSize.sp,
                color = Color.White
            )
        }
    }
}

/**
 * Locked Feature Badge - Badge per funzionalità bloccate
 */
@Composable
fun LockedFeatureBadge(
    modifier: Modifier = Modifier,
    size: BadgeSize = BadgeSize.MEDIUM,
    showLabel: Boolean = true
) {
    Row(
        modifier = modifier
            .background(
                color = Color.White.copy(alpha = 0.1f),
                shape = RoundedCornerShape(size.cornerRadius.dp)
            )
            .border(
                width = 1.dp,
                color = Color(0xFFFFD700).copy(alpha = 0.5f),
                shape = RoundedCornerShape(size.cornerRadius.dp)
            )
            .padding(
                horizontal = size.horizontalPadding.dp,
                vertical = size.verticalPadding.dp
            ),
        horizontalArrangement = Arrangement.spacedBy(size.spacing.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = Icons.Default.Lock,
            contentDescription = "Locked",
            tint = Color(0xFFFFD700),
            modifier = Modifier.size(size.iconSize.dp)
        )

        if (showLabel) {
            Text(
                text = "Premium",
                fontFamily = SoraFontFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = size.fontSize.sp,
                color = Color.White.copy(alpha = 0.8f)
            )
        }
    }
}

/**
 * Limit Reached Badge - Mostra il limite raggiunto
 */
@Composable
fun LimitReachedBadge(
    current: Int,
    max: Int,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .background(
                color = Color(0xFFFF8C00).copy(alpha = 0.2f),
                shape = RoundedCornerShape(8.dp)
            )
            .border(
                width = 1.dp,
                color = Color(0xFFFF8C00).copy(alpha = 0.5f),
                shape = RoundedCornerShape(8.dp)
            )
            .padding(horizontal = 10.dp, vertical = 5.dp),
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = Icons.Default.Warning,
            contentDescription = "Limit reached",
            tint = Color(0xFFFF8C00),
            modifier = Modifier.size(12.dp)
        )

        Text(
            text = "$current/$max",
            fontFamily = SoraFontFamily,
            fontWeight = FontWeight.SemiBold,
            fontSize = 12.sp,
            color = Color.White
        )
    }
}
